test/common/network/dns_impl_test.cc

```diff
@@ -43,19 +44,36 @@ namespace {
 typedef std::list<std::string> IpList;
 // Map from hostname to IpList.
 typedef std::unordered_map<std::string, IpList> HostMap;
+// Map from hostname to CNAME
+typedef std::unordered_map<std::string, std::string> CNameMap;
 // Represents a single TestDnsServer query state and lifecycle. This implements
 // just enough of RFC 1035 to handle queries we generate in the tests below.
 enum record_type { A, AAAA };
 
 class TestDnsServerQuery {
 public:
-  TestDnsServerQuery(ConnectionPtr connection, const HostMap& hosts_A, const HostMap& hosts_AAAA)
-      : connection_(std::move(connection)), hosts_A_(hosts_A), hosts_AAAA_(hosts_AAAA) {
+  TestDnsServerQuery(ConnectionPtr connection, const HostMap& hosts_A, const HostMap& hosts_AAAA,
+                     const CNameMap& cnames)
+      : connection_(std::move(connection)), hosts_A_(hosts_A), hosts_AAAA_(hosts_AAAA),
+        cnames_(cnames) {
     connection_->addReadFilter(Network::ReadFilterSharedPtr{new ReadFilter(*this)});
   }
 
   ~TestDnsServerQuery() { connection_->close(ConnectionCloseType::NoFlush); }
 
+  // Utility to encode a dns string in the rfc format. Example: \004some\004good\006domain
+  // RFC link: https://www.ietf.org/rfc/rfc1035.txt
+  static std::string encodeDnsName(const std::string& input) {
+    auto name_split = StringUtil::splitToken(input, ".");
+    std::string res;
+    for (const auto& it : name_split) {
+      res += static_cast<char>(it.size());
```

Comment:
Can you toss in an assert on the total size, just to make it super clear it won't work with illegally long DNS names?

Reply:
Will do